src/scripts/main.js

```diff
@@ -1,3 +1,129 @@
 'use strict';
 
-// write your code here
+function keyListener(e) {
+  if (game.getStatus() !== 'playing') {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowDown':
+      game.moveDown();
+      break;
+    case 'ArrowLeft':
+      game.moveLeft();
+      break;
+    case 'ArrowRight':
+      game.moveRight();
+      break;
+    case 'ArrowUp':
+      game.moveUp();
+      break;
+  }
+
+  update();
+}
+
+function update() {
+  game.getState().forEach((row, i) => {
+    row.forEach((el, j) => {
+      let cellCssClass = 'field-cell';
+
+      cellCssClass += el >= 2 ? ` field-cell--${el}` : '';
+      gameCells[i][j].innerText = el || '';
+      gameCells[i][j].className = cellCssClass;
+    });
```

Comment:
The nested forEach loops used to update the game state are not the most efficient way to handle DOM updates. Consider using a DocumentFragment to update the DOM in batches or updating only the cells that have changed to improve performance.